crates/swc_css_minifier/src/compressor/calc_sum.rs

```diff
@@ -120,6 +121,163 @@ impl CalcSumContext {
         }
     }
 
+    fn remove_zeroes(&mut self) {
+        if self.expressions.len() == 1 {
+            // We do not want to transform "calc(0)" into "calc()", that would be invalid
+            return;
+        }
+
+        let mut idx = 0;
+        while idx < self.expressions.len() {
+            if let Some(CalcProductOrOperator::Product(calc_product)) = self.expressions.get(idx) {
+                if CalcSumContext::is_calc_product_zero(calc_product)
+                    && self.try_to_remove_sum_operator_and_term(idx)
+                {
+                    continue;
+                }
+            }
+            idx += 1;
+        }
+    }
+
+    fn try_to_remove_sum_operator_and_term(&mut self, term_index: usize) -> bool {
+        if term_index == 0 {
+            if self.expressions.len() > 1 {
+                // If the next operator is minus ("-"), we try to merge it into its following
+                // term eg: calc(0 - 3% + 10px) => calc(-3% + 10px)
+                // ... but it's not always possible: calc(0 - pi)
+                let can_be_removed = match &self.expressions[1] {
+                    CalcProductOrOperator::Operator(CalcOperator {
+                        value: CalcOperatorType::Sub,
+                        ..
+                    }) => self.try_to_switch_sum_term_sign(term_index + 2),
+                    _ => true,
+                };
+
+                if can_be_removed {
+                    // Remove the term
+                    self.expressions.remove(term_index);
+                    // Remove the next operator (the sign has been merged into its term)
+                    self.expressions.remove(term_index);
+                }
+                return can_be_removed;
+            }
+            false
+        } else {
+            // Remove the term
+            self.expressions.remove(term_index);
+            // Remove the operator
+            self.expressions.remove(term_index - 1);
+            true
+        }
+    }
```

Comment:
Please read my comment above, we should not add extra loops, these things can be solved in one loop and expression, any sum or minus with `0` just return value itself or value with anither sign